content/components/action-bar.mdx

```diff
@@ -74,7 +105,16 @@ Dividers can be added to visually group related buttons.
 
 ### Overflow menu
 
-When the buttons don't fit in the available space, an overflow button ("kebab" icon) is added at the end of the action bar signaling that there are more actions available. Clicking on the overflow button opens a menu with the remaining actions that didn't fit.
+{/* <!-- TODO: determine how/if we want to support hiding items out of order --> */}
```

Comment:
I'm wondering if we want to support the ability to _always_ hide specific items. This was explored in an earlier design iteration that was abandoned.

Should we start to support this now? Or keep the behavior as-is?

If we do choose to support this, how should we let users specify actions that are always hidden?

1. Rely on the order the buttons are passed in, and let users specify a maximum number of action bar buttons
2. Let consumers specify which actions should go in the overflow menu regardless of what order they're passed in

Reply:
I spent some time exploring this functionality. Please excuse (or enjoy) the lo-fi ASCII art mocks 😅 

## Truncate after max number of items

Assume the max number of items was set to `5`.

### Option A

Limit how many buttons can be shown and rely on order. Any buttons that come after the limit are placed in the overflow menu.

#### Wide screen

```
[a] [b] | [c] [d] | [e] [...]
    |-----------------------|
    | [f]                   |
    | [g]                   |
    | [h]                   |
    | [i]                   |
    |-----------------------|
```

#### Narrow screen

```
[a] [b] | [c] [...]
    |-------------|
    | [d]         |
    | ------------|
    | [e]         |
    | [f]         |
    | [g]         |
    | [h]         |
    | [i]         |
    |-------------|
```

## Allow items to be explicitly overflowed in any order

Assume the buttons were passed in alphabetical order.

Assume that anything that appears in the overflow menu in a "wide screen" has been explicitly set to only appear in the overflow menu.

### Option A - use divider position to prepend to section:

#### Wide screen

```
[a] [b] | [e] [g] [i] [...]
    |---------------------|
    | [c]                 |
    | [d]                 |
    |---------------------|
    | [f]                 |
    | [h]                 |
    |---------------------|
```

#### Narrow screen

```
[a] [b] | [e] [...]
    |-------------|
    | [c]         |
    | [d]         |
    | ------------|
    | [g]         | <- prepended "g"
    | [i]         | <- prepended "i"
    | [f]         |
    | [h]         |
    |-------------|
```

### Option B (use divider position and order of button child for order of menu item):

#### Wide screen

```
[a] [b] | [e] [g] [i] [...]
    |---------------------|
    | [c]                 |
    | [d]                 |
    |---------------------|
    | [f]                 |
    | [h]                 |
    |---------------------|
```

#### Narrow screen

```
[a] [b] | [e] [...]
    |-------------|
    | [c]         |
    | [d]         |
    | ------------|
    | [f]         |
    | [g]         | <- added "g" to section based on the order it was passed
    | [h]         |
    | [i]         | <- added "i" to section based on the order it was passed
    |-------------|
```

Reply:
Are there usecases where certain items always show up in the overflow menu. I would say the comment box actionbar mostly does "Truncate after max number of items" and that too based on available space. If all items can be shown on the action bar then we simply show them. As the container reduces in space, we pop them on to the overflow menu.

Reply:
This is really cool! Do we already have examples from production or recipes where we do this?

> I'm wondering if we want to support the ability to always hide specific items.

That's interesting! I don't think I've come across this yet. (Or maybe, just never noticed it)

Reply:
Another question here, I assume for 99% of the use cases, we would want to follow the same order in wide and narrow (like in comment box and ListView). But, there could be scenarios where that's not true. For example, always keeping the destructive action in the end, but keeping it visible.

<img width="400" alt="icon buttons in action bar, destructive button is always visible as the last item" src="https://github.com/primer/design/assets/1863771/86160865-a626-4aeb-bf5d-1d1914f8fec4">

I think we can could come up with a custom prop that keeps the 99% case still pretty, something like `ActionBar.Icon icon={TrashIcon} avoidOverflow={true}`

Reply:
For the use case / customisation where the order outside the overflow isn't the same as the order inside the overflow: That's interesting! I don't think I've come across this yet. (Or maybe, just never noticed it)

I didn't fully understand the examples with divider here. 🤔 Are we inferring information based on the divider or is there explicit information about order passed here.

Reply:
> We could support something like <ActionBar.IconButton icon={TrashIcon} avoidOverflow={true} /> which would keep this icon visible until there are multiple avoidOverflow competing for space and then we start overflowing the last one.

> I think we should only pick one of these for the API, either avoidOverflow or preferOverflow, not both at the same time!

I like this functionality - it makes a lot of sense. I'm not sure if this is in scope though. What do you think @pksjce ?

I think we should go with `avoidOverflow` since the default value would be `false`.

---

> I didn't fully understand the examples with divider here. 🤔 Are we inferring information based on the divider or is there explicit order passed to ActionBar?

The only information we're inferring based on the ActionBar divider is that there should be a divider in the same spot in the ActionMenu.

Reply:
> @siddharthkp We could support something like `<ActionBar.IconButton icon={TrashIcon} avoidOverflow={true} />` which would keep this icon visible until there are multiple `avoidOverflow` competing for space and then we start overflowing the last one.
> @mperrotti I think we should go with `avoidOverflow` since the default value would be `false`.

I like this option too. ❤️ The example with the global nav kinda needs this behavior if it wants to use `ActionBar`.